### DISCUSSION TITLE
a noob feel confuse with the build progress.

### DISCUSSION THREAD

**Discussion by wushuaixue** [stm32-rs/stm32f1xx-hal#492] [Q&A] [answered]
To someone who will concern.

Recently, I try to use the rust for my stm32 coding. I found this lib and try to use that. everything goes well except the debug part.  I will show that at the blew.

PS D:\CODE\RUST\STMf103Pro> gdb -q target/thumbv7em-none-eabihf/debug/STMf103Pro
target/thumbv7em-none-eabihf/debug/STMf103Pro: No such file or directory.
warning: File "D:\CODE\RUST\STMf103Pro\.gdbinit" auto-loading has been declined by your `auto-load safe-path' set to "$debugdir:$datadir/auto-load".       
To enable execution of this file add
        add-auto-load-safe-path D:\CODE\RUST\STMf103Pro\.gdbinit
line to your configuration file "D:\Cadence\SPB_Data/.gdbinit".
To completely disable this security protection add
        set auto-load safe-path /
line to your configuration file "D:\Cadence\SPB_Data/.gdbinit".
For more information about this security protection see the
"Auto-loading safe path" section in the GDB manual.  E.g., run from the shell:
        info "(gdb)Auto-loading safe path"
(gdb) target remote :3333  
Remote debugging using :3333
warning: while parsing target description (at line 4): Target description specified unknown architecture "arm"
warning: Could not load XML target description; ignoring
warning: No executable has been specified and target does not support
determining executable automatically.  Try using the "file" command.
Truncated register 16 in remote 'g' packet
(gdb) load
No executable file specified.
Use the "file" or "exec-file" command.

by the way, if I dont want to debug, trans to load the .bin to the f103c8t6 directly using STM32 ST-LINK Utility. It remind me that File size is bigger than the flash memory size. I know that because c8t6 only have 64kb flash, but my release file have 138kb. if I have some wrong with the cargo build part. 

Thanks for your information. by the way, I am not native speaker in English, hope that issue will not bother you too much. Thanks again！

**Comment by burrbull** [comment, marked as answer]
1. Please, do not spam Github Issues with such questions. It it for reporting bugs/proposals. (converted to Discussion)
2. The size of `.elf` file is not the size of firmware. It includes advanced information to simplify access and debug, but not uploaded to your chip. To make a bare `.bin` from it you can use [`cargo objcopy`](https://github.com/rust-embedded/cargo-binutils?tab=readme-ov-file#objdump) or just [`cargo size`](https://github.com/rust-embedded/cargo-binutils?tab=readme-ov-file#size) to size firmware size.
3. We use [`probe-rs`](https://github.com/probe-rs/probe-rs/) tool instead of `openocd` and `gdb`. It is much more easy to use.
4. https://github.com/burrbull/stm32-template/ could help you to start.
5. It is better to discuss in matrix rooms: https://github.com/rust-embedded/awesome-embedded-rust?tab=readme-ov-file#community-chat-rooms
